File: Java/src/main/java/org/sisuns/condicionales/avanzado/Ejercicio_05.java
package org.sisuns.condicionales.avanzado;

public class Ejercicio_05 {
    public static void main(String[] args) {
        // Leer el consumo mensual en minutos (puedes cambiar el valor)
        int consumoMensual = 900;

        // Inicializar variables de precios y descuento
        double precioNormal = 0.03;
        double precioDescuento = 0.02;
        double descuentoPorcentaje = 1.25;

        // Calcular el costo según la política de la empresa
        double costoTotal;

        if (consumoMensual > 800) {
            // Calcular el costo para los primeros 300 minutos a precio normal
            double costoPrimeros300 = 300 * precioNormal;

            // Calcular el costo para los siguientes minutos a precio de descuento
            double costoRestantes = (consumoMensual - 300) * precioDescuento;

            // Calcular el descuento
            double descuento = (costoPrimeros300 + costoRestantes) * (descuentoPorcentaje / 100);

            // Calcular el costo final con descuento
            costoTotal = costoPrimeros300 + costoRestantes - descuento;
        } else if (consumoMensual > 500) {
            // Calcular el costo para los primeros 300 minutos a precio normal
            double costoPrimeros300 = 300 * precioNormal;

            // Calcular el costo para los siguientes minutos a precio de descuento
            double costoRestantes = (consumoMensual - 300) * precioDescuento;

            // Calcular el costo total sin descuento
            costoTotal = costoPrimeros300 + costoRestantes;
        } else if (consumoMensual > 300) {
            // Calcular el costo para todos los minutos a precio normal
            costoTotal = consumoMensual * precioNormal;
        } else {
            // Calcular el costo para todos los minutos a precio normal
            costoTotal = consumoMensual * precioNormal;
        }

        // Mostrar el costo total
        System.out.println("El costo total es: " + costoTotal);
    }
}

File: Python/Condicional/Avanzado/Ejercicio_05.py
# Leer el consumo mensual en minutos
consumo_mensual = int(input("Ingrese el consumo mensual en minutos: "))

# Inicializar variables de precios y descuento
precio_normal = 0.03
precio_descuento = 0.02
descuento_porcentaje = 1.25

# Calcular el costo según la política de la empresa
if consumo_mensual > 800:
    # Calcular el costo para los primeros 300 minutos a precio normal
    costo_primeros_300 = 300 * precio_normal

    # Calcular el costo para los siguientes minutos a precio de descuento
    costo_restantes = (consumo_mensual - 300) * precio_descuento

    # Calcular el descuento
    descuento = (costo_primeros_300 + costo_restantes) * (descuento_porcentaje / 100)

    # Calcular el costo final con descuento
    costo_total = costo_primeros_300 + costo_restantes - descuento
elif consumo_mensual > 500:
    # Calcular el costo para los primeros 300 minutos a precio normal
    costo_primeros_300 = 300 * precio_normal

    # Calcular el costo para los siguientes minutos a precio de descuento
    costo_restantes = (consumo_mensual - 300) * precio_descuento

    # Calcular el costo total sin descuento
    costo_total = costo_primeros_300 + costo_restantes
elif consumo_mensual > 300:
    # Calcular el costo para todos los minutos a precio normal
    costo_total = consumo_mensual * precio_normal
else:
    # Calcular el costo para todos los minutos a precio normal
    costo_total = consumo_mensual * precio_normal

# Mostrar el costo total
print("El costo total es:", costo_total)
